core/templates/dev/head/components/state-directives/response-header/response-header.directive.ts

```diff
@@ -56,7 +56,10 @@ angular.module('oppia').directive('responseHeader', [
             StateInteractionIdService, INTERACTION_SPECS) {
           var ctrl = this;
           ctrl.EditabilityService = EditabilityService;
-          ctrl.isInQuestionMode = StateEditorService.isInQuestionMode;
+
+          ctrl.isInQuestionMode = function() {
+            return StateEditorService.isInQuestionMode();
+          };
```

Comment:
@aks681 would you mind explaining why this change is needed? The code looks fine; this is just for my own learning.

Reply:
The original method was returning undefined, maybe due to the fact that `StateEditorService`  wasn't initialised by the time this directive was called, and hence `isInQuestionMode` was initialised to undefined and never changed. 
By doing this, since we are calling `StateEditorService` itself when this function is called, we get the correct value since it would get defined at some point.
At least, I think this is what is happening, since in the earlier case, the `isInQuestionMode` variable was undefined and it started working with this change :).